src/Application/GameLuaBindings.cpp

```diff
@@ -196,3 +385,29 @@ Character *getCharacterByIndex(int characterIndex) {
     logger->warning("Invalid character index. Asked for: {} but the party size is: {}", characterIndex, pParty->pCharacters.size());
     return nullptr;
 }
+
+sol::table createCharacterConditionTable(sol::state_view &luaState, const Character &character) {
+    sol::table result = luaState.create_table();
+
+    // I would have preferred to iterate over the internal IndexedArray keeping track of all the conditions but it's private.
+    for (int i = (int)Condition::CONDITION_FIRST; i < (int)Condition::CONDITION_LAST; ++i) {
```

Comment:
Add `Segment<Condition> allConditions()` to `CharacterEnumFunctions.h`, return `{CONDITION_FIRST, CONDITION_LAST}` from there (note that you don't need `Condition::` prefix), use it here.